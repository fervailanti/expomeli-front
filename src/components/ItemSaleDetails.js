import React from 'react'

const ItemSaleDetails = ({ image, condition, soldQty, title, price }) => (
  <div className="sale">
    <img src={image} alt="product" className="product-img" />
    <div className="details">
      <div>
        <p className="hairline">{`${condition} | ${soldQty} vendidos`}</p>
        <p className="title">{title}</p>
        <p className="price">{price}</p>
      </div>
      <div className="flex flex-col items-center mb-8">
        <button className="btn mt-4">Comprar ahora</button>
        <button className="btn-ghost mt-4">Agregar al carrito</button>
      </div>
    </div>
  </div>
)

export default ItemSaleDetails
