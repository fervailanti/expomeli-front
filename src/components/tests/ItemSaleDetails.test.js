import React from 'react'
import ItemSaleDetails from '../ItemSaleDetails'
import { render } from '@testing-library/react'

it('ItemSaleDetails component should render correctly', () => {
  const tree = render(
    <ItemSaleDetails
      image="https://picsum.photos/500"
      condition="Nuevo"
      soldQty={5}
      title="iPhone 11"
      price="$ 157.880,80"
    />
  )
  expect(tree.container).toMatchSnapshot()
})
