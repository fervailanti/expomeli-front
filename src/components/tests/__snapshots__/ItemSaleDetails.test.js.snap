// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSaleDetails component should render correctly 1`] = `
<div>
  <div
    class="sale"
  >
    <img
      alt="product"
      class="product-img"
      src="https://picsum.photos/500"
    />
    <div
      class="details"
    >
      <div>
        <p
          class="hairline"
        >
          Nuevo | 5 vendidos
        </p>
        <p
          class="title"
        >
          iPhone 11
        </p>
        <p
          class="price"
        >
          $ 157.880,80
        </p>
      </div>
      <div
        class="flex flex-col items-center mb-8"
      >
        <button
          class="btn mt-4"
        >
          Comprar ahora
        </button>
        <button
          class="btn-ghost mt-4"
        >
          Agregar al carrito
        </button>
      </div>
    </div>
  </div>
</div>
`;
